build/tools/Automation/Python/Python.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Oryx.Automation.Client;
+using Microsoft.Oryx.Automation.Models;
+using Microsoft.Oryx.Automation.Python.Models;
+using Microsoft.Oryx.Automation.Services;
+using Microsoft.Oryx.Automation.Telemetry;
+using Newtonsoft.Json;
+using Oryx.Microsoft.Automation.Python;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Microsoft.Oryx.Automation.Python
+{
+    public class Python
+    {
+        private readonly IHttpClient httpClient;
+        private readonly ILogger logger;
+        private readonly IVersionService versionService;
+        private readonly IYamlFileReaderService yamlFileReaderService;
+
+        public Python(IHttpClient httpClient, ILogger logger, IVersionService versionService, IYamlFileReaderService yamlFileReaderService)
+        {
+            this.httpClient = httpClient;
+            this.logger = logger;
+            this.versionService = versionService;
+            this.yamlFileReaderService = yamlFileReaderService;
+        }
+
+        public async Task RunAsync(string oryxRootPath)
+        {
+            List<VersionObj> versionObjs = await this.GetNewVersionObjsAsync();
+            if (versionObjs.Count > 0)
+            {
+                // Deserialize constants.yaml
+                string constantsYamlAbsolutePath = Path.Combine(oryxRootPath, "build", Constants.ConstantsYaml);
+                List<YamlConstants> yamlConstantsObjs = await this.yamlFileReaderService.ReadConstantsYamlFileAsync(constantsYamlAbsolutePath);
+
+                this.UpdateOryxConstantsForNewVersions(versionObjs, yamlConstantsObjs, oryxRootPath);
+            }
+        }
+
+        public async Task<List<VersionObj>> GetNewVersionObjsAsync()
+        {
+            var url = "https://www.python.org/api/v2/downloads/release/";
+            var response = await this.httpClient.GetDataAsync(url);
+            var releases = JsonConvert.DeserializeObject<List<Models.Release>>(response);
+
+            url = "https://oryx-cdn.microsoft.io/python?restype=container&comp=list&include=metadata";
+            HashSet<string> oryxSdkVersions = await this.httpClient.GetOryxSdkVersionsAsync(url);
+
+            var versionObjs = new List<VersionObj>();
+            foreach (var release in releases)
+            {
+                string newVersion = release.Name.Replace("Python", string.Empty).Trim();
+
+                if (!release.PreRelease &&
+                    this.versionService.IsVersionWithinRange(newVersion, minVersion: "3.10.9") &&
+                    !oryxSdkVersions.Contains(newVersion))
+                {
+                    Console.WriteLine($"newVersion {newVersion}");
```

Comment:
Are we able to add more context to this message that will be written to the console?